The problem im encountering is there seem to be problems with the `libusb` library. OpenEmbedded has 2 related recipes:
- [libusb](https://git.openembedded.org/meta-openembedded/tree/meta-oe/recipes-support/libusb/libusb-compat_0.1.5.bb?h=dunfell) - libusb-0.1 compatibility layer for libusb1
- [libusb1](https://layers.openembedded.org/layerindex/recipe/534/) - A cross-platform library to access USB devices.

The [DediProgSW/SF100Linux](https://github.com/DediProgSW/SF100Linux/blob/master/usbdriver.h#L6) includes `<libusb-1.0/libusb.h>` and requires `libusb-1.0` as a dependancy.

Yet adding either of them to [DEPENDS](https://docs.yoctoproject.org/ref-manual/variables.html#term-DEPENDS) puts out an error:

```bitbake
ERROR: dediprog-flasher-1.0+AUTOINC+e691f2d432-r0 do_compile: oe_runmake failed
ERROR: dediprog-flasher-1.0+AUTOINC+e691f2d432-r0 do_compile: Execution of '/build/tmp/work/armv7vet2hf-neon-rte-linux-gnueabi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/temp/run.do_compile.
312' failed with exit code 1:
  CC  dpcmd.o
  CC  usbdriver.o
  CC  FlashCommand.o
  CC  SerialFlash.o
In file included from usbdriver.c:1:
usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
    6 | #include <libusb-1.0/libusb.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:43: usbdriver.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from dpcmd.c:7:
usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
    6 | #include <libusb-1.0/libusb.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:43: dpcmd.o] Error 1
In file included from FlashCommand.c:6:
usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
    6 | #include <libusb-1.0/libusb.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:43: FlashCommand.o] Error 1
In file included from SerialFlash.c:5:
usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
    6 | #include <libusb-1.0/libusb.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:43: SerialFlash.o] Error 1
WARNING: exit code 1 from a shell command.

ERROR: Logfile of failure stored in: /build/tmp/work/armv7vet2hf-neon-rte-linux-gnueabi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/temp/log.do_compile.312
Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: make -j 16
|   CC  dpcmd.o
|   CC  usbdriver.o
|   CC  FlashCommand.o
|   CC  SerialFlash.o
| In file included from usbdriver.c:1:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: usbdriver.o] Error 1
| make: *** Waiting for unfinished jobs....
| In file included from dpcmd.c:7:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: dpcmd.o] Error 1
| In file included from FlashCommand.c:6:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: FlashCommand.o] Error 1
| In file included from SerialFlash.c:5:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: SerialFlash.o] Error 1
| ERROR: oe_runmake failed
| WARNING: exit code 1 from a shell command.
| ERROR: Execution of '/build/tmp/work/armv7vet2hf-neon-rte-linux-gnueabi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/temp/run.do_compile.312' failed with exit code 1:
|   CC  dpcmd.o
|   CC  usbdriver.o
|   CC  FlashCommand.o
|   CC  SerialFlash.o
| In file included from usbdriver.c:1:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: usbdriver.o] Error 1
| make: *** Waiting for unfinished jobs....
| In file included from dpcmd.c:7:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: dpcmd.o] Error 1
| In file included from FlashCommand.c:6:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: FlashCommand.o] Error 1
| In file included from SerialFlash.c:5:
| usbdriver.h:6:10: fatal error: libusb-1.0/libusb.h: No such file or directory
|     6 | #include <libusb-1.0/libusb.h>
|       |          ^~~~~~~~~~~~~~~~~~~~~
| compilation terminated.
| make: *** [Makefile:43: SerialFlash.o] Error 1
| WARNING: exit code 1 from a shell command.
|
ERROR: Task (/repo/recipes-devtools/dediprog-flasher/dediprog-flasher.bb:do_compile) failed with exit code '1'
```


Inside the `bitbake dediprog-flasher.bb -c devshell` i can see the `dpkg --list` output and files available at `$PKG_CONFIG_PATH`

```sh
$ dpkg --list | grep usb
$
```

```sh
$ echo $PKG_CONFIG_PATH
/build/tmp/work/armv7vet2hf-neon-rte-linux-gnuethe abi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/recipe-sysroot/usr/lib/pkgconfig:
/build/tmp/work/armv7vet2hf-neon-rte-linux-gnueabi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/recipe-sysroot/usr/share/pkgconfig
$ ls /build/tmp/work/armv7vet2hf-neon-rte-linux-gnueabi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/recipe-sysroot/usr/share/pkgconfig| grep usb
$ ls /build/tmp/work/armv7vet2hf-neon-rte-linux-gnueabi/dediprog-flasher/1.0+AUTOINC+e691f2d432-r0/recipe-sysroot/usr/lib/pkgconfig | grep usb
libusb-1.0.pc
libusb.pc
$ cat libusb.pc
prefix=/usr
exec_prefix=/usr
libdir=/lib
includedir=/usr/include
                                                                                                                                                                                              
emulated=1
emulated_by=libusb-1.0
                                                                                                                                                                                              
Name: libusb
Description: USB access library (libusb-1.0 compat wrapper)
Version: 0.1.12
Requires.private: libusb-1.0
Libs: -L${libdir} -lusb
Cflags: -I${includedir}

$ cat libusb-1.0.pc
prefix=/usr
exec_prefix=/usr
libdir=/lib
includedir=/usr/include
                                                                                                                                                                                              
Name: libusb-1.0
Description: C API for USB device access from Linux, Mac OS X, Windows, OpenBSD/NetBSD and Solaris userspace
Version: 1.0.22
Libs: -L${libdir} -lusb-1.0
Libs.private: -ludev  -pthread
Cflags: -I${includedir}/libusb-1.0
```

Adding the `-native` suffix didn't yield any valid results. 

The recipe is a copy of [imx-usb-loader_git.bb](https://github.com/3mdeb/meta-rte/blob/dediprog/recipes-devtools/imx-usb-loader/imx-usb-loader_git.bb) that also includes [<libusb-1.0/libusb.h>](https://github.com/boundarydevices/imx_usb_loader/blob/master/imx_usb.c#L35)

--- a/recipes-devtools/dediprog-flasher/dediprog-flasher.bb
+++ b/recipes-devtools/dediprog-flasher/dediprog-flasher.bb
@@ -6,7 +6,10 @@ LIC_FILES_CHKSUM = "file://LICENSE;md5=a23a74b3f4caf9616230789d94217acb"
 DEPENDS = "libusb"
 
 SRCREV = "e691f2d432144e3dbc82e9e0eea1ebaed4f3becf"
-SRC_URI = "git://github.com/DediProgSW/SF100Linux.git;protocol=https"
+SRC_URI = " \
+    git://github.com/DediProgSW/SF100Linux.git;protocol=https \
+    file://0001-add-support-for-cross-compilation.patch \
+    "
 
 PV = "1.0+${SRCPV}"
 

--- a/recipes-devtools/dediprog-flasher/dediprog-flasher.bb
+++ b/recipes-devtools/dediprog-flasher/dediprog-flasher.bb
@@ -1,0 +1,19 @@
+DESCRIPTION = "Linux software for Dediprog SF100 and SF600 SPI flash programmers"
+SECTION = "devel"
+LICENSE = "GPL-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=a23a74b3f4caf9616230789d94217acb"
+
+DEPENDS = "libusb"
+
+SRCREV = "e691f2d432144e3dbc82e9e0eea1ebaed4f3becf"
+SRC_URI = "git://github.com/DediProgSW/SF100Linux.git;protocol=https"
+
+PV = "1.0+${SRCPV}"
+
+S = "${WORKDIR}/git"
+
+do_install () {
+    oe_runmake DESTDIR=${D} install
+}
+
+inherit pkgconfig

--- a/recipes-devtools/dediprog-flasher/dediprog-flasher/0001-add-support-for-cross-compilation.patch
+++ b/recipes-devtools/dediprog-flasher/dediprog-flasher/0001-add-support-for-cross-compilation.patch
@@ -1,0 +1,20 @@
+Author: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
+Date:   Fri Jul 5 16:39:13 2024 +0200
+
+    Makefile: allow overriding CC
+    
+    Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
+
+diff --git a/Makefile b/Makefile
+index 139e5c0c3c3c..226303a6921e 100644
+--- a/Makefile
++++ b/Makefile
+@@ -12,7 +12,7 @@ endif
+ endif
+ 
+ PROGRAM = dpcmd
+-CC      = gcc
++CC     ?= gcc
+ PREFIX ?= /usr/local
+ 
+ PKG_CONFIG ?= pkg-config

--- a/recipes-extended/packagegroups/packagegroup-rte.bb
+++ b/recipes-extended/packagegroups/packagegroup-rte.bb
@@ -89,6 +89,7 @@ RDEPENDS_packagegroup-rte-stm = " \
 # packages useful for testing coreboot on various platforms
 RDEPENDS_packagegroup-rte-coreboot = " \
     flashrom \
+    dediprog-flasher \
     ifdtool \
     cbfstool \
     "

